src/Akeneo/Platform/Bundle/InstallerBundle/Persistence/Sql/GetInstallDatetime.php

```
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Platform\Bundle\InstallerBundle\Persistence\Sql;
+
+use DateTimeImmutable;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @author JM Leroux <jean-marie.leroux@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class GetInstallDatetime
+{
+    public function __construct(private readonly Connection $connection)
+    {
+    }
+
+    public function __invoke(): ?\DateTime
+    {
+        $sql = <<< SQL
+            SELECT `values` FROM pim_configuration WHERE code = 'install_data';
+        SQL;
+
+        $values = $this->connection->executeQuery($sql)->fetchOne();
+
+        if (false === $values) {
+            return null;
```

IMO, it seems more logic to throw an exception instead of returning null because we expect to get a value, getting null is an error case